docs/src/pages/theming.mdx

```diff
@@ -0,0 +1,430 @@
+  sure to import default stylesheet into your app.
+</Callout>
+
+```ts filename="tailwind.config.ts"
+import { withUt } from "@uploadthing/react/tw";
+
+/** @type {import('tailwindcss').Config} */
+export default withUt({
+  // your config goes here
+});
+```
+
+```js filename="tailwind.config.js"
+// @ts-check
+const { withUt } = require("@uploadthing/react/tw");
+
+/** @type {import('tailwindcss').Config} */
+module.exports = withUt({
+  // your config goes here
+});
+```
+
+<Callout type="warning">
+  Note: Once you extended your config, it is recommended to remove default
+  stylesheet import
+</Callout>
+
+`withUt` does two major things:
+
+- Adds new entry to `content` property so that it includes UploadThing's
+  components. It allows to avoid having duplicated styles in your bundle.
+- Registers plugin and adds nice-to-have classes to improve styling experience.
+
+#### `className` prop
+
+`className` prop accepts any classes so you can pass there anything you like.
+When it comes to TailwindCSS, to make styling easier, `withUt` adds few utility
+classes and variants for you to leverage.
+
+Tailwind variants that allow applying styles to internal elements and their
+arbitrary variant alternatives:
+
+| Custom variant       |                        Arbitrary variant |
+| :------------------- | ---------------------------------------: |
+| `ut-button`          |          `&>*[data-ut-element="button"]` |
+| `ut-allowed-content` | `&>*[data-ut-element="allowed-content"]` |
+| `ut-label`           |           `&>*[data-ut-element="label"]` |
+| `ut-upload-icon`     |     `&>*[data-ut-element="upload-icon"]` |
+
+Tailwind classes that allow applying styles based on state of element and their
+arbitrary `data` class alternatives:
+
+| Custom class   |          Arbitrary class |
+| :------------- | -----------------------: |
+| `ut-readying`  |  `data-[state=readying]` |
+| `ut-ready`     |     `data-[state=ready]` |
+| `ut-uploading` | `data-[state=uploading]` |
+
+These variants and classes can be used in conjunction with each other to make
+component match your design in the exact way you want.
+
+```jsx
+<UploadButton
+    className="bg-blue-500 ut-button:bg-cyan-500 ut-button:ut-readying:bg-cyan-500/50"
+               ^           ^ applied to button   ^ applied to button when loading config
+               | applied to container
+
+/>
+
+<UploadDropzone
+    className="bg-blue-500 ut-label:text-lg ut-allowed-content:ut-uploading:text-red-300"
+               ^           ^                ^ applied to allowed content when uploading file
+               |           | applied to label
+               | applied to container
+
+/>
+```
+
+#### `appearance` prop
+
+If you need, you can pass TailwindCSS classes directly to specific elements of
+component or provide a callback that will be called with current state of the
+component and will return a string
+
+```jsx
+<UploadButton
+  appearance={{
+    button({ ready, isUploading }) {
```

Comment:
Why should I use the callback here instead of tailwind variants?

Reply:
Well, it would definitely make more sense for a user to leverage TW variants instead of callback for state-specific styling. I guess I can remove usage of callback in this particular example to avoid promotion of not the best practices